### ISSUE TITLE
Sage Pay failure if first name is too long

### ISSUE THREAD

**Issue by wmortada** [eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor#213]
While doing some testing on a site using this extension with Sage Pay I came across an issue if the billing first name is too long. Apparently Sage Pay has a limit of 20 characters for the `BillingFirstname`. This is probably fine for many names but there are some cases where names will exceed this. The problem is that this isn't handled very well from the user's perspective.

The only message that I got as a user was:

> Sorry, there was an error processing your payment. Please try again later.

Looking at the error log I could see more detail but this isn't shown to the user:

```
Aug 25 14:55:37  [alert] failed processor transaction omnipay_SagePay_Server
Array
(
    [INVALID] => 3110 : The BillingFirstnames value is too long.
)
```

It looks like this is an error returned from Sage Pay, but it isn't a great user experience as it doesn't tell the user what the error is or how long this field should be.

I imagine that this is an issue with other fields (and possibly other payment processors?) so I guess this is an issue that needs to be fixed in this extension and/or in OmniPay?

Ideally the extension should provide better feedback to the end user if the input fails validation rules. The user should also be able to correct the input so that it meets the validation rules.

**Comment by kurund** [comment]
@eileenmcnaughton 

We are thinking it would be good to add some js validation that will prevent the user error. If you agree can work on a PR for the same.

**Comment by eileenmcnaughton** [comment]
@kurund is this closed by #219?

**Comment by kurund** [comment]
@eileenmcnaughton 

Yes, we can close this issue.
